src/java.base/share/classes/sun/security/rsa/RSAPadding.java

```diff
@@ -236,24 +234,22 @@ public int getMaxDataSize() {
     }
 
     /**
-     * Pad the data and return the padded block.
+     * Pad the data and return the result or null if error occurred.
      */
-    public byte[] pad(byte[] data) throws BadPaddingException {
+    public byte[] pad(byte[] data) {
         return pad(data, 0, data.length);
     }
 
     /**
-     * Pad the data and return the padded block.
+     * Pad the data and return the result or null if error occurred.
      */
-    public byte[] pad(byte[] data, int ofs, int len)
-            throws BadPaddingException {
+    public byte[] pad(byte[] data, int ofs, int len) {
         if (len > maxDataSize) {
-            throw new BadPaddingException("Data must be shorter than "
-                + (maxDataSize + 1) + " bytes but received "
-                + len + " bytes.");
+            return null;
         }
         switch (type) {
         case PAD_NONE:
+            // assert len == paddedSize and data.length - ofs > len?
```

Comment:
I think this assert is only relevant on tip, where the output of this function is passed to Output.pass(). However, it's in the jdk21u backport, so I guess we should leave it in for consistency....

Reply:
I can't seem to find the Output.pass() reference I was looking at previously. I would leave this comment in for consistency with the tip and 21u PRs.